src/components/widgets/ColorscalePicker.js

```diff
@@ -1,41 +1,95 @@
-import ColorscalePicker, {Colorscale} from 'react-colorscales';
+import ColorscalePicker, {
+  Colorscale,
+  COLOR_PICKER_CONSTANTS,
+} from 'react-colorscales';
+import Dropdown from './Dropdown';
+import Info from '../fields/Info';
 import PropTypes from 'prop-types';
-import React, {Component} from 'react';
+import React, {Component, Fragment} from 'react';
 
 class Scale extends Component {
-  constructor(props) {
-    super(props);
+  constructor() {
+    super();
+
     this.state = {
+      selectedColorscaleType: 'sequential',
       showColorscalePicker: false,
     };
-    this.toggle = this.toggle.bind(this);
+
+    this.onChange = this.onChange.bind(this);
+    this.onClick = this.onClick.bind(this);
   }
 
-  toggle() {
+  onClick() {
     this.setState({
       showColorscalePicker: !this.state.showColorscalePicker,
     });
   }
 
+  onChange(selectedColorscaleType) {
+    this.setState({selectedColorscaleType});
+  }
+
   render() {
-    const {selected, onColorscaleChange} = this.props;
+    const {onColorscaleChange, selected} = this.props;
+    const {selectedColorscaleType, showColorscalePicker} = this.state;
+    const description =
+      COLOR_PICKER_CONSTANTS.COLORSCALE_DESCRIPTIONS[selectedColorscaleType];
+    const colorscaleOptions = COLOR_PICKER_CONSTANTS.COLORSCALE_TYPES.map(
+      type => ({
+        label: type + ' scale',
```

Comment:
actually now that I look at it, could we make this `scales` with an `s` please? :)